### PR TITLE
Rebuild town center

### PR DESCRIPTION
**Pull request by lanfranconi**
I brought back the `tcMonitor` rule to make explorer units rebuild the town center if there's no more town center.

### CODE REVIEW

File: AI3/aiMain.xs [hunk 1]
```diff
@@ -5347,6 +5295,110 @@ minInterval 10
...
+    }
+    default:
+    {
+        aiPlanAddUnitType(buildPlan, cUnitTypeExplorer, 1, 1, 1);
+        break;
+    }
+    }
+
+    // Instead of base ID or areas, use a center position and falloff.
+    aiPlanSetVariableVector(buildPlan, cBuildPlanCenterPosition, 0, kbBaseGetLocation(cMyID, kbBaseGetMainID(cMyID)));
```
Comment (alistair-jah) [inline AI3/aiMain.xs:5370]: Radius should be fixed to something like 80~100 meters, which is roughly twice the surface of the default TC LOS. We don't want TC's to be built too far from the main base, even in the case where the build plans fails frequently.

**Review by alistair-jah — CHANGES_REQUESTED** [review]
Just a tiny little bit more tweaks and you're good to go 👍

File: AI3/aiMain.xs [hunk 2]
```diff
@@ -5347,6 +5295,110 @@ minInterval 10
...
+    aiPlanSetVariableInt(buildPlan, cBuildPlanBuildingTypeID, 0, cUnitTypeTownCenter);
+    // Priority.
+    aiPlanSetDesiredPriority(buildPlan, 100);
+    // Mil vs. Econ.
+    aiPlanSetMilitary(buildPlan, false);
+    aiPlanSetEconomy(buildPlan, true);
+    // Escrow.
+    aiPlanSetEscrowID(buildPlan, cEconomyEscrowID);
+    // Builders.
+    switch (kbGetCiv())
```
[the remark this answers is not in the document]
Comment (lanfranconi) [reply AI3/aiMain.xs:5328]: Some campaign heroes can't build anything, so this section makes sense. But I guess all these explorer units should be added to the plan regardless of the civilization. I'm going to do that, actually.

[the remark this answers is not in the document]
Comment (alistair-jah) [reply AI3/aiMain.xs]: Ah, yeah, I forgot lol

Alright, good job 👌 I'm gonna add you directly as a contributor and co-author. What do you think?

**Review by alistair-jah — APPROVED** [review]

**Pull request merged**